build.sbt

```diff
@@ -67,13 +67,23 @@ lazy val downloadWeb3AndInstallStd = taskKey[Unit]("Download alephium-web3 sourc
 downloadWeb3AndInstallStd := {
   val web3Dir = new File(s"target/web3/alephium-web3-${Version.web3}")
   val stdDir = new File("compiler-access/src/main/resources/std")
+  val vscodeStdDir = new File("plugin-vscode/out/std")
   val web3StdDir = new File(s"target/web3/alephium-web3-${Version.web3}/packages/web3/std")
 
   if(java.nio.file.Files.notExists(web3Dir.toPath())) {
       IO.unzipURL(new URL(s"https://github.com/alephium/alephium-web3/archive/refs/tags/v${Version.web3}.zip"), new File("target/web3"))
   }
 
   IO.copyDirectory(web3StdDir, stdDir)
+  IO.copyDirectory(web3StdDir, vscodeStdDir)
+  setReadOnly(vscodeStdDir)
+}
+
+def setReadOnly(directory: File): Unit = {
+  directory.setReadOnly()
+  val files = directory.listFiles()
+  if (files != null)
+    files foreach (_.setReadOnly())
```

Comment:
you'll see me coming :) but it then only works for vscode. In my case, my nvim correctly tried to open a file in my home folder: 
![image](https://github.com/alephium/ralph-lsp/assets/2979182/4b654c13-17c4-4104-bab1-b2f3848a2678)
but the file isn't there.

Should we try the `metals` approach? When you want to jump to a dependency file, it grabs it from the jar file and keep it in the project itself: 
![image](https://github.com/alephium/ralph-lsp/assets/2979182/38a58cae-13da-425f-8883-ee2c5dd24f09)

![image](https://github.com/alephium/ralph-lsp/assets/2979182/5a43c33e-5c39-411f-ad9d-6c5e343b6122)

We could still store the files in `RALPH_LSP_HOME`, but the files could be fetch on the fly when needed, we check if they are present, if not, take them from `ralph-lsp.jar`.

Open question as always, but we need to be careful of having an "editor agnostic" lsp server. I'm using nvim, I know Hongchao is using emacs and will soon try the server, so it would be nice to work for him directly too

Reply:
Thinking out loud, if `RALPH_LSP_HOME` is defined, we could load std files there, otherwise we keep them in project  folder. So it works even if user didn't defined a home folder

Reply:
Yep that makes sense. Was kinda deferring the whole dependency management thing until #44. `RALPH_LSP_HOME` would've been temporary.

> Should we try the metals approach?

Metals uses Coursier, ivy, maven etc for dependency management which gets complicated. Copying std files into VSCode plugin was the easy solution for now, this way there is no IO or version management to worry about for new releases.

> we need to be careful of having an "editor agnostic" lsp server.

Yes. Server has always been editor agnostics. Just as std files gets copied into vscode's plugin by `sbt`, the same could be done for any editor or plugin. It's not specific to just vscode. For example, I'm an Intellij user, doing the same for IntelliJ plugin would be another line of sbt `IO.copyDirectory(web3StdDir, intelliJStdDir)` [here](https://github.com/alephium/ralph-lsp/blob/d8cb8a9807d75a43d6eb81c010f0af2e1dcacd77/build.sbt#L78). I guess our support for nvim is not a plugin yet.

> files could be fetch on the fly when needed

To have this, I can implemented a small portion of #44 involving dependency management now as a prerequisite for this PR.

Reply:
but we do have the std files inside our `ralph-lsp.jar` no?
Could we extract them and put them in the `RALPH_LSP_HOME` when we start the server? 
Without doing dependency management, just copy what we have in the jar file.